### PR TITLE
feat(resources/main): disable controls

### PR DESCRIPTION
**Pull request by itschip**
CPU msec goes brrrr

### CODE REVIEW

**Comment by RockySouthpaw** [comment]
It would probably be best to disable all control actions and enable a select amount instead. Controls 32-35 (WASD) for movement, 249 (N) for push to talk, and 23 (F)for entering a vehicle. 

With the current configuration assuming I’m reading it correctly, It looks like you’re still able to open the weapon wheel, punch (would probably happen on left clicks), shoot, and other actions that probably shouldn’t be possible while holding the phone. 

You would then just need to enable all controls again once you’re done. 

I can conduct some testing later and make sure but just from looking over the code and my own testing previously, these evwnts will most likely occur.

**Comment by RockySouthpaw** [comment]
Tested in game and it's as described above.

https://user-images.githubusercontent.com/55056068/113645626-332e7000-9655-11eb-93c4-53b2191764c0.mp4

Will need to go about this another way than you are currently.

**Comment by RockySouthpaw** [comment]
Yea it looks like once you do `DisableAllControlActions(0);` you cant enable them from then on 🤔 So the best way would be an obnixious serious of disabling controls lol

File: resources/client/cl_main.ts [hunk 1]
```diff
@@ -196,3 +200,30 @@ on(`__cfx_nui:${PhoneEvents.CLOSE_PHONE}`, async (data: any, cb: Function) => {
...
+      if (!isPhoneOpen) {
+        clearTick(loopTick);
+        return;
+      }
+
+      if (isPhoneOpen) {
+        DisableControlAction(0, 1, true);
+        DisableControlAction(0, 2, true);
+        DisableControlAction(0, 16, true);
+        DisableControlAction(0, 17, true);
```
Comment (RockySouthpaw) [inline resources/client/cl_main.ts:218]:
```suggestion
        DisableControlAction(0, 17, true); //    Select Previous Weapon
        DisableControlAction(0, 22, true); //    Jump
        DisableControlAction(0, 24, true); //    Attack
        DisableControlAction(0, 25, true); //    Aim 37
        DisableControlAction(0, 37, true); //    Weapon Wheel
```

File: resources/client/cl_main.ts [hunk 2]
```diff
@@ -196,3 +200,30 @@ on(`__cfx_nui:${PhoneEvents.CLOSE_PHONE}`, async (data: any, cb: Function) => {
...
+
+  const loopTick = setTick(
+    async (useTimeout = false): Promise<void> => {
+      if (!isPhoneOpen) {
+        clearTick(loopTick);
+        return;
+      }
+
+      if (isPhoneOpen) {
+        DisableControlAction(0, 1, true);
```
Comment (RockySouthpaw) [inline resources/client/cl_main.ts:215]:
```suggestion
        DisableControlAction(0, 1, true);  //    Look Left Right
```

File: resources/client/cl_main.ts [hunk 3]
```diff
@@ -196,3 +200,30 @@ on(`__cfx_nui:${PhoneEvents.CLOSE_PHONE}`, async (data: any, cb: Function) => {
...
+        DisableControlAction(0, 2, true);
```
Comment (RockySouthpaw) [inline resources/client/cl_main.ts:216]:
```suggestion
        DisableControlAction(0, 2, true);  //    Look Up Down
```

File: resources/client/cl_main.ts [hunk 4]
```diff
@@ -196,3 +200,30 @@ on(`__cfx_nui:${PhoneEvents.CLOSE_PHONE}`, async (data: any, cb: Function) => {
...
+        DisableControlAction(0, 16, true);
```
Comment (RockySouthpaw) [inline resources/client/cl_main.ts:217]:
```suggestion
        DisableControlAction(0, 16, true); // 	 Select Next Weapon
```

**Comment by RockySouthpaw** [comment]
@TasoOneAsia can this be closed since it’s being done differently now? I assume the current edits are essentially null and void.

**Comment by TasoOneAsia** [comment]
Yes can be closed as this and any other frame ticks should probably be handled in lua to keep that precious resource time number down.

**Pull request closed without merging**